pkg/virt-controller/watch/vmi.go

```diff
@@ -584,26 +628,43 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, SuccessfulCreatePodReason, "Created virtual machine pod %s", pod.Name)
 		return nil
+	} else {
+		if isTempPod(pod) && !isWaitForFirstConsumer {
+			// DV is no longer in WaitForFirstConsumer phase, means that tempPod has done its job and should be deleted
+			vmiKey := controller.VirtualMachineKey(vmi)
+			c.podExpectations.ExpectDeletions(vmiKey, []string{controller.PodKey(pod)})
+			err := c.clientset.CoreV1().Pods(vmi.Namespace).Delete(pod.Name, &v1.DeleteOptions{})
+			if err != nil && !k8serrors.IsNotFound(err) {
+				c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedDeletePodReason, "Failed to delete temporary pod %s", pod.Name)
+				c.podExpectations.DeletionObserved(vmiKey, controller.PodKey(pod))
+				return &syncErrorImpl{fmt.Errorf("Failed to delete temporary pod: %v", err), FailedDeletePodReason}
+			}
+			c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, SuccessfulDeletePodReason, "Deleted temporary pod %s", pod.Name)
+			return nil
+		}
 	}
 	return nil
 }
 
-func (c *VMIController) handleSyncDataVolumes(vmi *virtv1.VirtualMachineInstance, dataVolumes []*cdiv1.DataVolume) (bool, syncError) {
+func (c *VMIController) handleSyncDataVolumes(vmi *virtv1.VirtualMachineInstance, dataVolumes []*cdiv1.DataVolume) (bool, bool, syncError) {
 
 	ready := true
+	wffc := false
 
 	for _, volume := range vmi.Spec.Volumes {
 		// Check both DVs and PVCs
 		if volume.VolumeSource.DataVolume != nil {
 			for _, dataVolume := range dataVolumes {
 				if dataVolume.Name == volume.VolumeSource.DataVolume.Name {
-					if dataVolume.Status.Phase != cdiv1.Succeeded {
+					if dataVolume.Status.Phase == cdiv1.WaitForFirstConsumer {
+						wffc = true
```

Comment:
how that PVCs can be referenced instead of the datavolume, don't we need to check for wffc in the event the PVC is referenced?

https://github.com/kubevirt/kubevirt/pull/4277

Reply:
Right, after merging #4277, kubevirt correctly works with WFFC DV, but it does not work if the PVC is used instead of a DV.
It can be handled, but then we would need to follow the owner and get the original DV (so basically do everything that was hidden in the new utility function Is cdiv1.IsPopulated). We are already working on this.

Reply:
@brybacki what should be the expectation for this PR. are we waiting for the cdi change to land so this pr can leverage it?

Reply:
Yes: https://github.com/kubevirt/containerized-data-importer/pull/1442

Should I mark this as on hold until new cdi version can be used?

Reply:
yup, makes sense to me.

Reply:
updated, PVC's are no covered

Reply:
@davidvossel can you check again? new stuff is in last commit `check PVC if it waits for first consumer …`